Marlin/Marlin_main.cpp

```diff
@@ -9483,25 +9483,39 @@ inline void gcode_M999() {
 }
 
 #if ENABLED(SWITCHING_EXTRUDER)
-  inline void move_extruder_servo(uint8_t e) {
-    const int angles[2] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-    MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-    safe_delay(500);
+  inline void move_extruder_servo(const uint8_t e) {
+    stepper.synchronize();
+    constexpr int16_t angles[] = SWITCHING_EXTRUDER_SERVO_ANGLES;
+    static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");
+
+    #if EXTRUDERS & 1
```

Comment:
@thinkyhead , maybe I'm wrong, but
with `SWITCHING_EXTRUDER `we have only one stepper per two tools.
And in this void `e `equal to tool 0,1,3,4 or 5.
with 2 `EXTRUDERS `we will have only one `E_STEPPERS`.
with 3 or 4 `EXTRUDERS `- two `E_STEPPERS`.

So we can't use `e` with `E_STEPPERS `here.